docs/devguide/dtypes.rst

```diff
@@ -1,2 +1,179 @@
 Implementing a New Data Type
 ============================
+
+COMPAS data types are classes that are based on :class:`compas.data.Data`.
+
+Data types can be serialized to JSON with
+
+* :func:`compas.json_dump`
+* :func:`compas.json_dumps`
+* :func:`compas.json_dumpz`
+
+and deserialized with the corresponding "load" functions
+
+* :func:`compas.json_load`
+* :func:`compas.json_loads`
+* :func:`compas.json_loadz`
+
+All geometry objects and data structures,
+and also, for example, the visualization scene,
+are serializable data types.
+
+
+Creating a new data type
+========================
+
+In most cases, to create a new data type it is sufficient to implement the ``__data__`` property
+of your custom data class.
+
+.. code-block:: python
+
+    class CustomData(Data):
+
+        def __init__(self, a, b, name=None)
+            super(CustomData, self).__init__(name=name)
+            self.a = a
+            self.b = b
+
+        @property
+        def __data__(self):
+            data = super(CustomData, self).__data__
+            data['a'] = self.a
+            data['b'] = self.b
+            return data
+
+
+>>> custom = CustomData(a=1, b=2)
+>>> compas.json_dump(custom, "custom.json")
+>>> result = compas.json_load("custom.json")
+>>> isinstance(result, CustomData)
+True
+>>> result.a
+1
+>>> result.b
+2
+
+If the attributes stored in the data dictionary defined by the ``__data__`` property
+are different from the initialization parameters of the class,
+you also have to customize the ``__from_data__`` class method to compensate for the difference.
```

Comment:
I don't know if the Internet Best Practice is a thing here, but there is this [Key words for use in RFCs to Indicate Requirement Levels ](https://www.ietf.org/rfc/rfc2119.txt) I find very clear.

For example here, 
```suggestion
you must also customize the ``__from_data__`` class method to compensate for the difference.
```